--- seabios.submodule/src/acpi.c    2012-04-15 08:27:40.000000000 +0600
+++ seabios.submodule/src/acpi.c    2012-05-05 01:52:07.000000000 +0600
@@ -212,6 +212,11 @@
 
 #include "acpi-dsdt.hex"
 
+#define CONFIG_OEM_SLIC
+#ifdef CONFIG_OEM_SLIC
+#include "acpi-slic.hex"
+#endif
+
 static void
 build_header(struct acpi_table_header *h, u32 sig, int len, u8 rev)
 {
@@ -224,6 +229,10 @@
     h->oem_revision = cpu_to_le32(1);
     memcpy(h->asl_compiler_id, CONFIG_APPNAME4, 4);
     h->asl_compiler_revision = cpu_to_le32(1);
+#ifdef CONFIG_OEM_SLIC
+    if (sig == RSDT_SIGNATURE)    // only RSDT is checked by win7 & vista
+      memcpy(h->oem_id, ((struct acpi_table_header*)SLIC)->oem_id, 14);
+#endif
     h->checksum -= checksum(h, len);
 }
 
@@ -709,6 +718,17 @@
     ACPI_INIT_TABLE(build_srat());
     ACPI_INIT_TABLE(build_pcihp());
 
+#ifdef CONFIG_OEM_SLIC
+    { void *buf = malloc_high(sizeof(SLIC));
+      if (!buf)
+        warn_noalloc();
+      else {
+        memcpy(buf, SLIC, sizeof(SLIC));
+        ACPI_INIT_TABLE(buf);
+      }
+    }
+#endif
+
     u16 i, external_tables = qemu_cfg_acpi_additional_tables();
 
     for (i = 0; i < external_tables; i++) {
